### PR TITLE
percona: update to 8.0.20-11

### PR DESCRIPTION
**Pull request by chrstphrchvz**
Remove obsolete pre-activate

Remove `+openssl` variant
(OpenSSL no longer optional, yaSSL no longer bundled)

#### Description
https://www.percona.com/doc/percona-server/LATEST/release-notes/Percona-Server-8.0.20-11.html

###### ~~Tested on~~

**Untested.**

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by chrstphrchvz** [comment]
Cc @MarcusCalhoun-Lopez and @yan12125, who most recently updated this port.

Also cc @herbygillot: since you maintain `mysql8`, would you be interested in maintaining this port?

The upstream Percona server 8.0.x is based on MySQL 8.0.x releases, but the portfiles for each have diverged noticeably (both descend from much older MySQL versions), and more work would be needed to reconcile them. I wonder if one port should be retired in favor of the other, to reduce maintenance effort. The `percona` port has been around for longer and would appear to have more dependents; there are far more `+percona` port variants than `+mysql8` port variants.

**Comment by chrstphrchvz** [comment]
The build currently fails; I believe the router component needs to be disabled, as was done for `mysql8`.

File: databases/percona/Portfile [hunk 1]
```diff
@@ -130,7 +130,7 @@ if {$subport eq $name} {
                         -DENABLE_DTRACE:BOOL=OFF \
                         -DWITH_EDITLINE:STRING=bundled \
                         -DWITH_LIBWRAP:BOOL=ON \
-                        -DWITH_SSL:STRING=bundled \
+                        -DWITH_SSL:STRING=${prefix} \
```
Comment (chrstphrchvz) [inline databases/percona/Portfile:133]: Should this instead be `-DWITH_SSL:PATH="${prefix}"` or does it not matter?

**Comment by herbygillot** [comment]
> Cc @MarcusCalhoun-Lopez and @yan12125, who most recently updated this port.
> 
> Also cc @herbygillot: since you maintain `mysql8`, would you be interested in maintaining this port?
> 
> The upstream Percona server 8.0.x is based on MySQL 8.0.x releases, but the portfiles for each have diverged noticeably (both descend from much older MySQL versions), and more work would be needed to reconcile them. I wonder if one port should be retired in favor of the other, to reduce maintenance effort. The `percona` port has been around for longer and would appear to have more dependents; there are far more `+percona` port variants than `+mysql8` port variants.

I haven't looked into Percona, but I don't think I agree with retiring one of the `mysql8` or `percona` ports in favor of the other.  Vanilla MySQL should continue to be an option alongside Percona for those who want it.  Are there far more `percona` variants because `mysql8` was not available in MacPorts for quite some time?  Again I'm not sure if that's the case, I haven't taken the time to dig in.

**Comment by chrstphrchvz** [comment]
> Are there far more `percona` variants because `mysql8` was not available in MacPorts for quite some time?

That is the impression I have: unlike how each new major version of MySQL ended up in a separate port, followed by adding new dependent variants in other ports, the `percona` port was instead upgraded in-place and existing variants left alone. (Though many ports only have variants for older MySQL versions, and the effort migrating them to newer versions has taken many years: [trac#43431](https://trac.macports.org/ticket/43431))

**Comment by chrstphrchvz** [comment]
Next build error to fix is:
```
[  2%] Building C object extra/zstd/CMakeFiles/zstd.dir/lib/decompress/zstd_decompress.c.o
⁝
error: use of undeclared identifier 'ZSTD_FRAMEHEADERSIZE_PREFIX'
```

File: databases/percona/Portfile (hunk 1 above)
Comment (herbygillot) [reply databases/percona/Portfile:134]: Yes, it's `${prefix}` in `mysql8`.

Comment (chrstphrchvz) [reply databases/percona/Portfile:134]: Both quoting and the use of `STRING` vs `PATH` (I do not know much at all about CMake). There is inconsistency among MySQL ports or various options in them.

**Comment by chrstphrchvz** [comment]
The failure on Xcode earlier than 9.0 is something `mysql8` avoids, though I might not understand exactly why. Upstream only supports compiling with Xcode &ge; 9.0 on macOS. The `mysql8` port uses `compiler.blacklist-append {clang < 900}` and disables the compiler check using `-DFORCE_UNSUPPORTED_COMPILER=ON`; I presume this is done to allow using compatible compilers other than Xcode, while still avoiding Xcode < 9.0. gcc5 and certain recent MacPorts' Clang versions might similarly be excluded. See discussion in #7774.

File: databases/percona/Portfile (hunk 1 above)
Comment (herbygillot) [reply databases/percona/Portfile:134]: Disclaimer: I am not a CMake expert either.

Basically CMake variables can have types, like `PATH`, `STRING`, or `BOOL`.  When in doubt, just specify the option and value without the `:<TYPE>` suffix.  The types themselves look like they are simply used to help CMake create the right graphical widgets when configuring using CMake's graphical tools: https://cmake.org/cmake/help/v3.0/command/set.html

If you're curious about what a particular variable's type is, you can dig into the files in the `cmake` directory, to see whether the variable is set using `OPTION()` or `SET()`.  If `OPTION`, then the variable's type is `BOOL`.  Else look for where the variable is being set using `SET`.  You may see it as `SET(<var> ... CACHE <type> ...)`.  If you see `SET(<var> ... INTERNAL ...)`, then that variable's type is `STRING`.

`CACHE` variables and `OPTION`s are the kind of variables CMake exposes on the command line, and which we would be setting in a `Portfile`.

Here are some examples from `percona-server` source:
* `WITH_SSL` set in `CACHE`: https://github.com/percona/percona-server/blob/41fa622c3399b68b77104054673fab49b92c3fb8/cmake/ssl.cmake#L98
* `WITH_LIBWRAP` set using `OPTION`: https://github.com/percona/percona-server/blob/41fa622c3399b68b77104054673fab49b92c3fb8/configure.cmake#L144

**Comment by chrstphrchvz** [comment]
There are no other changes I'm currently interesting in making.

@herbygillot I'd like to leave a note in both ports to suggesting `percona` be synchronized with `mysql8`, if that's alright with you.

**Comment by herbygillot** [comment]
> There are no other changes I'm currently interesting in making.
> 
> @herbygillot I'd like to leave a note in both ports to suggesting `percona` be synchronized with `mysql8`, if that's alright with you.

That's 100% fine with me.  The versions don't exactly match between them (`mysql8` is 8.0.21 where `percona` is 8.0.20-11), but that's fine.

**Comment by chrstphrchvz** [comment]
Rebased to resolve conflict with https://github.com/macports/macports-ports/commit/4080d7a1a4421c671eb19141211f7f1b01228149#diff-13b50e6b5bbf7d9b208184a02d35eea0

**Comment by herbygillot** [comment]
Hey @chrstphrchvz, just took a look.  I got it to build, but think we're missing a utility...

According to `percona-server`'s notes, we should run `$ sudo -u _mysql /opt/macports-test/lib/percona/bin/mysql_install_db` for a new installation, but unfortunately that utility is completely missing.

```
$ port contents percona | grep install_db
$
```

**Comment by chrstphrchvz** [comment]
@herbygillot Thanks for taking a closer look. I think the notes might be outdated: `mysql56` and newer have instead said to run `${prefix}/lib/${name_mysql}/bin/mysqld`. I would try what is suggested in `mysql8`, but I wonder if there are other missing/outdated things in this port that need to be addressed before the notes in `mysql8` would work.

**Comment by kencu** [comment]
I don't know where we are with this PR, but having "Draft" or "WIP" PRs sitting around in the queue for months and months is not helpful. Please close or fix!

**Comment by chrstphrchvz** [comment]
The next steps for this would be to incorporate the 8.0.22-13 update and sort out the notes issues Herby noticed.

Unfortunately this PR has been much more effort than I intended to put into this port. It is far easier for me to move on to smaller/shorter tasks. This one would have to wait until I (or someone who cares more about this port) get around to it.

@kencu if you're looking for old PRs to merge, I suggest #7955, #8205, #8206, #9196. I would also appreciate if someone with access to older macOS versions could test macports/macports-base#214.

**Comment by chrstphrchvz** [comment]
I will not be finishing this anytime soon. I believe this port should be removed if no one is around to care for it.

**Pull request closed without merging**